### PR TITLE
Fix array_read code error.

### LINKED ISSUE #15822: "array_read" API demo doesn't work
**System information**
-PaddlePaddle version: 1.2.1
-OS Platform and Distribution: Ubuntu 18.04
-Python version: Python 2.7.15rc1

**To Reproduce**
Use the code example provided in the following API doc:
http://paddlepaddle.org/documentation/docs/zh/1.2/api_cn/layers_cn.html#cn-api-fluid-layers-array-read

**Describe your current behavior**
```
Traceback (most recent call last):
  File "array_read.py", line 4, in <module>
    arr = fluid.layers.array_read(tmp,i=i)
  File "/usr/local/lib/python2.7/dist-packages/paddle/fluid/layers/control_flow.py", line 1103, in array_read
    raise TypeError("array should be tensor array vairable")
TypeError: array should be tensor array vairable
```

**Code to reproduce the issue**
```python
import paddle.fluid as fluid
tmp = fluid.layers.zeros(shape=[10],dtype='int32')
i = fluid.layers.fill_constant(shape=[1],dtype='int64',value=10)
arr = fluid.layers.array_read(tmp,i=i)
```

**Others**
BTW, there is a typo in error message, "vairable" should be "variable".

### PR DESCRIPTION
**Pull request by zhaoyuchen2018**
Fix array_read code example error.
fix https://github.com/PaddlePaddle/Paddle/issues/15822
test=develop
image

### CODE REVIEW

File: python/paddle/fluid/layers/control_flow.py [hunk 1]
```diff
@@ -941,9 +941,9 @@ def array_read(array, i):
     Examples:
         .. code-block:: python
 
-          tmp = fluid.layers.zeros(shape=[10], dtype='int32')
+          array = fluid.layers.create_array(dtype='float32')
```
Comment (panyx0718) [inline python/paddle/fluid/layers/control_flow.py:932]: is there any test to cover this issue?

**Review by panyx0718 — DISMISSED** [review]

File: python/paddle/fluid/layers/control_flow.py (hunk 1 above)
Comment (zhaoyuchen2018) [reply python/paddle/fluid/layers/control_flow.py:932]: no test as this is comment code for doc， will show in document

**Review by shanyi15 — DISMISSED** [review]

File: python/paddle/fluid/layers/control_flow.py [hunk 2]
```diff
@@ -929,9 +929,9 @@ def array_read(array, i):
...
           i = fluid.layers.fill_constant(shape=[1], dtype='int64', value=10)
-          arr = layers.array_read(tmp, i=i)
+          item = fluid.layers.array_read(array, i)
```
Comment (chengduoZH) [inline python/paddle/fluid/layers/control_flow.py:934]: Do you write data before calling `layers.array_read`?
Do you test the above code?

Comment (zhaoyuchen2018) [reply python/paddle/fluid/layers/control_flow.py:934]: code is in document, just show how to use array, so not fill data, and above code can run normally.

**Review by chengduoZH — APPROVED** [review]
The next PR should refine the test cast.

**Review by panyx0718 — APPROVED** [review]

**Review by shanyi15 — APPROVED** [review]

**Pull request merged**